server/text_generation_server/utils/gptq/exllamav2.py

```diff
@@ -167,7 +166,11 @@ def __init__(self, qweight, qzeros, scales, g_idx, bias, bits, groupsize):
         self.qweight = qweight
         self.qzeros = qzeros
         self.scales = scales
-        self.g_idx = g_idx
+
+        if torch.equal(g_idx, torch.tensor([i // groupsize for i in range(self.infeatures)], dtype=torch.int32, device=g_idx.device)):
+            self.g_idx = None
+        else:
+            self.g_idx = g_idx

```

Comment:
Fixes a bug in the original exllamav2 code.

Reply:
Wow THANKS